go-backend/db/organization.go

```diff
@@ -131,41 +146,47 @@ func (s *pgStore) CreateOrganization(ctx context.Context, org Organization) (cre
 		org.Hi5Limit,
 		org.Hi5QuotaRenewalFrequency,
 		org.Timezone,
-	).Scan(&lastInsertId)
+		org.CreatedAt,
+	).Scan(&lastInsertID)
 	if err != nil {
 		logger.WithField("err", err.Error()).Error("Error creating organization")
 		return
 	}
 
-	err = s.db.Get(&createdOrganization, getOrganizationQuery, lastInsertId)
-
+	err = s.db.Get(&createdOrganization, getOrganizationQuery, lastInsertID)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			// TODO: Log that we can't find the organization even though it's just been created
+		}
+	}
 	return
 }
 
 func (s *pgStore) UpdateOrganization(ctx context.Context, reqOrganization Organization, organizationID int) (updatedOrganization Organization, err error) {
-		var dbOrganization Organization
-		err = s.db.Get(&dbOrganization, getOrganizationQuery, organizationID)
-
-		_, err = s.db.Exec(
-			updateOrganizationQuery,
-			reqOrganization.Name,
-			reqOrganization.ContactEmail,
-			reqOrganization.DomainName,
-			reqOrganization.SubscriptionStatus,
-			reqOrganization.SubscriptionValidUpto,
-			reqOrganization.Hi5Limit,
-			reqOrganization.Hi5QuotaRenewalFrequency,
-			reqOrganization.Timezone,
-			organizationID,
-		)
-		if err != nil {
-			logger.WithField("err", err.Error()).Error("Error updating organization")
-			return
-		}
-
-		err = s.db.Get(&updatedOrganization, getOrganizationQuery, organizationID)
+	var dbOrganization Organization
+	err = s.db.Get(&dbOrganization, getOrganizationQuery, organizationID)
 
+	_, err = s.db.Exec(
+		updateOrganizationQuery,
+		reqOrganization.Name,
+		reqOrganization.ContactEmail,
+		reqOrganization.DomainName,
+		reqOrganization.SubscriptionStatus,
+		reqOrganization.SubscriptionValidUpto,
+		reqOrganization.Hi5Limit,
+		reqOrganization.Hi5QuotaRenewalFrequency,
+		reqOrganization.Timezone,
+		reqOrganization.CreatedAt,
+		organizationID,
+	)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error updating organization")
 		return
+	}
+
+	err = s.db.Get(&updatedOrganization, getOrganizationQuery, organizationID)
```

Comment:
you should add `err` handing here because in future somebody can add some code below this line without handling the error. So its always recommended to handle error and return

Reply:
Good catch, just plain missed that one. 46cfee73b5ffdc1dbd60f0fb8d2f55e0e81723be